File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        s=0
        e=0
        cnt=0
        ans=-(1<<31)
        while e<len(nums):
            if nums[e]==0:
                cnt=cnt+1
            if cnt<=k:
                ans=max(ans,e-s+1)
                e=e+1
            else:
                while cnt>k and s<=e:
                    if nums[s]==0:
                        cnt=cnt-1
                    s=s+1
                if cnt<=k:
                    ans=max(ans,e-s+1)
                e=e+1
        return ans
        

File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int length=nums.length;
        int left=0;
        int right=0;
        int zeroCount=0;
        int maxi=0;
        while(right<length){
            if(nums[right]==0){
                zeroCount++;
            }
            while(zeroCount>k){
                if(nums[left]==0){
                    zeroCount=zeroCount-1;
                }
                left=left+1;
            }
            if(zeroCount<=k){
                maxi=Math.max(maxi,right-left+1);
            }
            right++;
        }
        return maxi; 
    }
}


